src/utils/__tests__/ObjectUtils-test.js

```diff
@@ -102,27 +113,21 @@ describe('ObjectUtils', () => {
     it('handles immutable values if one input is falsy', () => {
       expect(() => shallowEqual(Map({ a: 1 }), null)).not.toThrow();
     });
-  });
-
-  describe('deepEqual', () => {
-    it('deeply compares values', () => {
-      expect(deepEqual(1, 1)).toBe(true);
-      expect(deepEqual(1, 0)).toBe(false);
-
-      expect(deepEqual(1, null)).toBe(false);
-      expect(deepEqual(1, undefined)).toBe(false);
-      expect(deepEqual(0, false)).toBe(false);
 
-      expect(deepEqual({}, null)).toBe(false);
-      expect(deepEqual({}, {})).toBe(true);
-
-      expect(deepEqual({ a: 1 }, {})).toBe(false);
-      expect(deepEqual({}, { a: 1 })).toBe(false);
-      expect(deepEqual({ a: 1 }, { a: 1 })).toBe(true);
-      expect(deepEqual({ a: 1 }, { a: 'different' })).toBe(false);
-      expect(deepEqual({ a: 1, b: { c: 2 } }, { a: 1, b: { c: 2 } })).toBe(
-        true
-      );
+    it('handles large immutables reasonably quickly', () => {
```

Comment:
It might also be worthing capturing this perf data somewhere. But that can come later.